embrace-android-sdk/src/main/cpp/file_writer.c

```diff
@@ -158,147 +165,313 @@ emb_error *emb_read_errors_from_file(const char *path) {
 }
 
 char *emb_crash_to_json(emb_crash *crash) {
+    if (crash == NULL) {
+        return NULL;
+    }
     EMB_LOGDEV("Starting serialization of emb_crash struct to JSON string.");
-    JSON_Value *root_value = json_value_init_object();
-    JSON_Object *root_object = json_value_get_object(root_value);
+    JSON_Value *root_value = NULL;
+    JSON_Object *root_object = NULL;
+    JSON_Value *crash_value = NULL;
+    JSON_Object *crash_object = NULL;
     char *serialized_string = NULL;
 
-    emb_add_metadata_to_json(crash, root_object);
-    emb_add_basic_info_to_json(crash, root_object);
-
-    // crash data
-    EMB_LOGDEV("Serializing crash data.");
-    JSON_Value *crash_value = json_value_init_object();
-    JSON_Object *crash_object = json_value_get_object(crash_value);
-
-    json_object_set_number(root_object, kUnwinderErrorCode, crash->unwinder_error);
-
-    emb_exception *exception = &crash->capture;
-    emb_add_exc_info_to_json(crash, crash_object, exception);
-
-    JSON_Value *frames_value = json_value_init_array();
-    JSON_Array *frames_object = json_value_get_array(frames_value);
-    emb_add_exc_to_json(exception, frames_object);
-
-    json_object_set_value(crash_object, kFramesKey, frames_value);
+    // initialize JSON objects.
+    root_value = json_value_init_object();
+    if (root_value == NULL) {
+        goto Exit;
```

Comment:
https://xkcd.com/292/ 😆

Reply:
No raptors yet! 🤞 

Goto might seem unorthodox but it's actually a pretty useful pattern for freeing up allocated memory within a C function. It's almost like a try-catch in Java, which is basically implemented as a goto under the hood...

Reply:
It looks really clean! But I had to share the comic lol